tools/clang/test/HLSLFileCheck/hlsl/preprocessor/pragma/pragma-diagnostic.hlsl

```diff
@@ -0,0 +1,26 @@
+// RUN: %dxc -T vs_6_0 %s | FileCheck %s
+
+float4 foo;
+
+// This function has no output semantic on purpose in order to produce an error,
+// otherwise, the warnings will not be captured in the output for FileCheck.
+float main() : OUT {
+  float4 x = foo;
+
+#pragma dxc diagnostic push
+
+// CHECK-NOT: equality comparison with extraneous parentheses
+#pragma dxc diagnostic ignored "-Wparentheses-equality"
+  if ((x.y == 0))
+  {
+
+// CHECK: error: implicit truncation of vector type
+#pragma dxc diagnostic error "-Wconversion"
+    return x;
+
+  }
+  return x.y;
+
+#pragma dxc diagnostic pop
```

Comment:
I guess we can trust that the clang pragma push and pop work. Otherwise, we might verify that warnings have been restored to their default after this

Reply:
It's a good point, and the thought did cross my mind before submitting the PR, but it didn't seem worth the extra effort for another test at the time.

Reply:
Fair enough. I defer to you. I just wanted to make sure it had been considered.